lib/logproto/logproto-text-server.c

```diff
@@ -377,31 +379,37 @@ log_proto_text_server_fetch_from_buffer(LogProtoBufferedServer *s, const guchar
                                         const guchar **msg, gsize *msg_len)
 {
   LogProtoTextServer *self = (LogProtoTextServer *) s;
-  const guchar *eol;
   LogProtoBufferedServerState *state = log_proto_buffered_server_get_state(&self->super);
   gboolean result = FALSE;
 
-  eol = log_proto_text_server_locate_next_eol(self, state, buffer_start, buffer_bytes);
+  const guchar *eol = log_proto_text_server_locate_next_eol(self, state, buffer_start, buffer_bytes);
 
-  if (!eol &&
-      ((buffer_bytes == state->buffer_size) ||
-       log_proto_buffered_server_is_input_closed(&self->super)))
+  gboolean message_reached_max_size = buffer_bytes >= self->super.super.options->max_msg_size;
```

Comment:
I think I would've extracted the condition into a separate function, possibly the prefix chopped off, to make reading the code easier:

```
if (!eol && (_message_size_too_large() || log_proto_buffered_server_is_input_closed(&self->super))
   ;
```
Wouldn't that be better?